BUGFIX: Use countAll of AssetRepository to get count of all assets

Nikdro opened the pull request:
To get count of all assets, we should use the AssetRepository.
Solves #2358

robertlemke commented:
Hi @Nikdro, thanks for your pull request! As far as I can see, this would break the support for asset sources. For example, if you have an external asset source configured (such as an enterprise DAM, Next Cloud or Google Drive), the calculated number would be incorrect.

Is there any specific error or so which made you creating this bugfix? Then we can look into what the root cause for that error ist.

Nikdro commented:
Hi @robertlemke,

thanks for your feedback, I got the point.

I adjusted the implementation. Now it's iterating over all available AssetSources to get the complete count.  
The Bug I fixed with this PR is described in #2358 (like mentioned in description)
